hawkular-alerts-engine/src/test/java/org/hawkular/alerts/engine/RulesEngineTest.java

```diff
@@ -1888,4 +1893,84 @@ public void checkEqualityInRulesEngine() throws Exception {
         assertTrue(rulesEngine.getFact(jsonfmt1c2) != null);
         assertTrue(rulesEngine.getFact(jsonfmt1c2eval) != null);
     }
+
+    @Test
+    public void missingDataTestNoData() throws Exception {
+        //Initial trigger and condition
+        Trigger t1 = new Trigger("tenant", "trigger-m", "Missing test Trigger");
+        MissingCondition fmt1 = new MissingCondition("tenant", "trigger-m", "data-id", 3000L);
+
+        t1.setEnabled(true);
+
+        // Missing state is generated on reloadTrigger()
+        missingStates.add(new MissingState(t1, fmt1));
+
+        rulesEngine.addFact(t1);
+        rulesEngine.addFact(fmt1);
+
+        // Rules Invoker thread updates missing states periodically
+        MissingState missingState = missingStates.iterator().next();
+
+        rulesEngine.removeFact(missingState);
+
+        // It simulates that we have not data in 4 seconds > of 3 seconds interval defined
+        missingState.setPreviousTime(missingState.getTime());
```

Comment:
missingState.previousTime should not be updated here.
This comes from a preliminar version.